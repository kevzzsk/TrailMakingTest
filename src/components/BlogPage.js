
import React, { Component } from 'react'

import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Stepper from '@material-ui/core/Stepper';
import Paper from '@material-ui/core/Paper';
import Step from '@material-ui/core/Step';
import StepLabel from '@material-ui/core/StepLabel';
import Typography from '@material-ui/core/Typography';
import Skeleton from "@material-ui/lab/Skeleton"

import BlogData from '../template/blogData'


/** Blog page for participants to learn more about dementia patient and help lines */
class BlogPage extends Component {

    /** @member */
    state = {
        activeStep: 10,
        steps: ["He/she should see a doctor for an assessment and diagnosis.",
        "Get some understanding about dementia. Your doctor can put you in touch with people who can help.",
        "Rally for support. Caring for people with dementia is challenging and can be exhausting both physically and emotionally. Learn to talk about your problems. Make sure that you get enough rest. Take care of your own physical and mental well-being.",
        "Group support is important. Meet with others who are also caring for people with dementia. Sharing provides mutual support for caregivers.",
        "Get expert advice especially with coping with challenging behaviour."],
        loading: false
    }

    /** Init Data */
    componentDidMount(){
        this.setState({loading:true})
        setTimeout(function() { //Start the timer
            this.setState({loading: false}) //After 1 second, set render to true
        }.bind(this), 1000)
    }

    /**
     * @method
     * @param {int} index index of ICON
     * @description return respective icon based on index
     */
    getIconLabel = (index)=>{
        switch (index) {
            case 1:
                return <i className="far fa-edit"/>
            case 2:
                return <i className="fas fa-info-circle"/>
            case 3:
                return <i className="fas fa-life-ring"/>
            case 4:
                return <i className="fas fa-users"/>
            case 5:
                return <i className="fas fa-user-md"/>
            default:
                break;
        }
    }

    /**
     * @method
     * @param {*} props 
     * @description return Icon
     * @returns {Object} Display ICON
     */
    ColorlibStepIcon =(props)=> {
      
        return (
          <div>
            {this.getIconLabel(props.icon)}
          </div>
        );
      }

    render() {
        return (
            <div className="blog-bg">
                <div>
                    <Typography className="m-3" variant="h3">Blog</Typography>
                    {this.state.loading? [1,2,3].map((item,i)=><Skeleton key={i} width="97%" height={460} className="m-3" />) :BlogData.map((blog, i) => {
                        return <Card key={i} className="m-3 blogpost" >
                            <CardActionArea onClick={() => window.open(blog.webUrl, "_blank")}>
                                <img className="card-img-top" style={{ objectFit: "cover", height: "300px" }} src={blog.coverImage} alt="imagea" ></img>
                                <CardContent>
                                    <Typography gutterBottom variant="h5" component="h2">
                                        {blog.title}
                                    </Typography>
                                    <Typography variant="body2" color="textSecondary" component="p">
                                        {blog.description}
                                    </Typography>
                                </CardContent>
                            </CardActionArea>
                            <CardActions>
                                <Button size="small" color="primary">
                                    Share
                                    </Button>
                                <Button size="small" color="primary" onClick={() => window.open(blog.webUrl, "_blank")}>
                                    Learn More
                                    </Button>
                            </CardActions>
                        </Card>
                    })})
                </div>
                <div className="m-2">
                    <Paper className="p-4">
                    <Typography variant="h4" style={{fontWeight:"bold",letterSpacing:"2px"}}>What can I do if someone I know is showing signs of dementia?</Typography>
                    <Stepper activeStep={this.state.activeStep} orientation="vertical">
                        {this.state.steps.map((label, index) => (
                            <Step key={label}>
                                <StepLabel StepIconComponent={this.ColorlibStepIcon} >{label}</StepLabel>
                            </Step>
                        ))}
                    </Stepper>
                    </Paper>
                    <Paper className="p-4 mt-3 card-left">
                        <i style={{fontSize:"40px",color:"rgb(248,107,70)"}} className="fas fa-phone-alt align-self-center"/>
                        <div className="ml-2">
                        <Typography variant="h5">HPB Dementia Infoline</Typography>
                        <Typography variant="h3" style={{color:"rgb(248,107,70)",fontWeight:"600"}}>1800 223 1123</Typography>
                        </div>
                    </Paper>
                    <Paper className="p-4 mt-3 card-left">
                        <i style={{fontSize:"40px",color:"rgb(248,107,70)"}} className="fas fa-phone-alt align-self-center"/>
                        <div className="ml-2">
                        <Typography variant="h5">Alzheimer's Disease Association Dementia Helpline</Typography>
                        <Typography variant="h3" style={{color:"rgb(248,107,70)",fontWeight:"600"}}>6377 0700</Typography>
                        </div>
                    </Paper>
                </div>
            </div>
        )
    }
}


export default BlogPage