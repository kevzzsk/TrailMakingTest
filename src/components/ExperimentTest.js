
import React, { Component } from 'react';
import BubbleScreen from './BubbleScreen';
import SideBar from './SideBar';
import Trail from '../template/testTemplate';

/**
 * ExperimentTest handles the interactive part of the experiment
 * It renders the test experiment page
 * @param {*} props
 * 
 */
class ExperimentTest extends Component {

    static defaultProps={
        activeStep:1,
        location:{
            state:{
                experimentID:0,
                participantDetails:{}
            }
        }
    }

    /** @constructor */
    constructor(props) {
        super(props);
        this.state = {
            id: null,
            ready: false,
            expIndex: 0,
            trail: Trail.experiment[0]
        }
    }
    render() {
        const {participantDetails, experimentID,activeStep,trail,doctorID} = this.props.location.state
        return (
            <div className="experiment-bg">
                <div className="item-exp">
                        <BubbleScreen trail={this.state.trail} />
                </div>
                <div className="item-side">
                    <SideBar goBack={this.props.history.goBack} doctorID={doctorID} trail={trail}  participantDetails={participantDetails}  activeStep={activeStep} id={experimentID} expIndex={this.state.expIndex} ready={this.state.ready} test={true}/>
                </div>
            </div>
        )
    }
}

export default ExperimentTest;