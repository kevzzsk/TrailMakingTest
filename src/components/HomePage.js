import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { FormControl, TextField, CircularProgress } from "@material-ui/core"
import Footer from './Footer'
import Appbar from './Appbar'
import axios from "axios"

/**
* 
* @param {Object} aboutRef Reference to About in home page 
*/
class HomePage extends Component {
    
    state = {
        id: "",
        error: "",
        loading: false
    }

    /**
     * @method
     * @description Handles submission of experiment ID
     * @param {Object} event DOM reference to the form
     */
    submitFormHandler = (event) => {
        event.preventDefault();
        console.log("SUBMIT");
        this.showLoader()
        axios.get(`https://cors-anywhere.herokuapp.com/https://easya.fyp2017.com/api/tmt/viewExperiment?experimentID=${this.state.id}`,{headers:{"X-Requested-With":"XMLHttpRequest"}})
            .then(res => {
                this.setState({
                    error: ""
                })
                this.hideLoader()
                this.props.history.push({
                    pathname: '/form', state: {
                        experimentID: this.state.id,
                        expIndex: 0,
                        activeStep: 0,
                        trails: res.data.templateExperiments
                    }
                })
            })
            .catch(err => {
                this.setState({
                    error: "Enter a valid ID"
                })
                this.hideLoader()
            })
    }

    /**
     * @method 
     * @description SetState to hide loader */
    hideLoader = () => {
        this.setState({ loading: false });
    }
    /** 
     * @method
     * @description SetState to show loader */
    showLoader = () => {
        this.setState({ loading: true });
    }
    /** 
     * @method
     * @description Handle input change when user enter experimentID 
     * @param {Object} e Event DOM
    */
    handleOnChange = (e) => {
        this.setState({
            [e.target.name]: e.target.value,
            error: ""
        })
    }

    render() {
        return (
            <div>
                <header className="header-bg">
                    <div className="head">
                        <h1>"Every 3 seconds someone in the world develops dementia"</h1>
                        <div>
                            <p>There were an estimated 50 million people worldwide living with dementia in 2017. This number will almost double every 20 years, reaching 75 million in 2030 and 131.5 million in 2050.
Much of the increase will be in low and middle income countries. Already 58% of people with dementia live in low and middle income countries, but by 2050 this will rise to 68%. </p>
                        </div>
                    </div>

                </header>
                <form id = "experiment_form" onSubmit={this.submitFormHandler}>
                    <div className="form-group experiment">
                        <br />
                        <FormControl variant="outlined" className="w-100">
                            <TextField
                                required
                                id="outlined-id"
                                label="Experiment ID"
                                placeholder="A0001"
                                name="id"
                                error={this.state.error.length > 0}
                                onChange={this.handleOnChange}
                                value={this.state.id}
                                margin="normal"
                                variant="outlined"
                                type="text"
                                rowsMax={1}
                            />
                        </FormControl>
                        {this.state.error.length > 0 && <small style={{ color: "#D8000C" }}>{this.state.error}</small>}
                        <small className="form-text text-muted">By continuing you comply with our terms and conditions.</small>
                        <div className="position-relative">
                            <button disabled={this.state.loading} type="submit" className="btn experiment-btn" id="submitButton" >Try Experiment</button>
                            {this.state.loading && <CircularProgress size={40} className="button-loading" id="loading"/>}
                        </div>
                    </div>
                </form>
                <section className="intro" ref={this.props.aboutRef}>
                    <h2>About Us</h2>
                    <div>
                        <p>We are a group of students from NTU who have developed an application based on Trail Making Tests for researchers and doctors alike to diagnose and treat patients with dementia.
                            Researchers are able to use our application for research purposes by getting participants to use our application.
                        </p>
                    </div>
                </section>


                <div className="top-services">
                    <h2>We Are Looking For</h2>
                    <div className="services">
                        <div className="service-one">
                            <p className="service-icon"><i className="fas fa-users"></i></p>
                            <p className="service-title">Participants</p>
                            <p>Participants of research groups to aid in the research of dementia.</p>
                        </div>
                        <div className="service-two">
                            <p className="service-icon"><i className="fas fa-user-graduate"></i></p>
                            <p className="service-title">Researchers</p>
                            <p>Scientists and researchers currently making great strides in identifying potential new ways to help diagnose, treat, and even prevent Alzheimer's and related dementias.</p>

                        </div>
                        <div className="service-three">
                            <p className="service-icon"><i className="fas fa-user-md"></i></p>
                            <p className="service-title">Doctors</p>
                            <p>Doctors trained in brain conditions (neurologist) or doctors trained to treat older adults (geriatrician).</p>

                        </div>
                    </div>
                </div>
                <Footer />
            </div>
        )
    }
}

export default HomePage;