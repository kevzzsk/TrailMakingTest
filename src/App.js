import React, { Component } from 'react';
import './App.css';

import { Route, Switch, withRouter } from "react-router-dom";

import ExperimentPage from './components/ExperimentPage';
import Appbar from './components/Appbar';
import HomePage from './components/HomePage';
import CompletionPage from './components/CompletionPage'
import LoginPage from './components/LoginPage';
import UserPage from './components/UserPage';
import CreateExperiment from './components/createNewExp/CreateExperiment';
import SignUpPage from './components/SignUpPage';
import ForgetPassPage from './components/ForgetPassPage';
import ExperimentTest from './components/ExperimentTest';
import ExperimentStat from './components/ExperimentStat';
import CreateExperimentForm from "./components/createNewExp/CreateExperimentForm"
import CreateCompleted from "./components/createNewExp/CreateCompleted"
import PersonalParticulars from "./components/PersonalParticulars"
import DoctorPage from './components/DoctorPage';
import CreateTemplate from './components/CreateTemplate'
import BlogPage from './components/BlogPage'
import SubmissionPage from './components/SubmissionPage'
import CreateSubmission from "./components/createNewExp/CreateSubmission"
import MyPatients from './components/MyPatients'
class App extends Component {
	/**
	 * 
	 * @param {*} props 
	 */

	/** @constructor */
	constructor(props) {
		super(props)
		window.onclick = () => {
			this.setState({
				pathname: window.location.pathname,
			})
		}
		this.state = {
			pathname: window.location.pathname,
			aboutRef: React.createRef(null),
			isAuthenticated: false
		}
	}

	/**
	 * @method
	 */
	componentDidMount() {
		this.setState({
			isAuthenticated: localStorage.getItem("isAuthenticated") === null ? false : localStorage.getItem("isAuthenticated") === "true"
		})
	}

	/**
	 * @method
	 * @description Handle Login function which set isAuthenticated to True and route to the correct account page. Doctor (accountType 1) Researcher (accountType 2) 
	 * @param {Object} acc Account object
	 * @returns {void}
	 */
	login = (acc) => {
		localStorage.setItem("isAuthenticated", "true")
		this.setState({
			isAuthenticated: true
		})
		if (acc.accountType === 1) {
			this.props.history.replace("/doctor-page",
				{
					account: acc
				})
		} else if (acc.accountType === 0) {
			console.log("REPLACE")
			this.props.history.replace(
				"/user-page",
				{
					account: acc
				}
			)
		}
	}

	/** 
	 * @method
	 * @description Handle logout button which set isAuthenticated to False and remove account data. Route back to /login
	 */
	logout = () => {
		localStorage.setItem("isAuthenticated", "false")
		localStorage.removeItem("account")
		this.setState({
			isAuthenticated: false
		})
		this.props.history.push("/login")
	}



	render() {
		return (
			<div id="bg">
				<Appbar pathname={this.state.pathname} aboutRef={this.state.aboutRef} isAuthenticated={this.state.isAuthenticated} logout={this.logout} history={this.props.history} ></Appbar>
				<Switch>
					<Route exact path="/" render={(props) => <HomePage {...props} aboutRef={this.state.aboutRef} />} />
					<Route path="/experiment" exact component={ExperimentPage} />
					<Route path="/completion" exact component={CompletionPage} />
					<Route path="/form" exact component={PersonalParticulars} />
					<Route path="/test" exact component={ExperimentTest} />
					<Route path="/submission" exact component={SubmissionPage} />
					<Route path="/login" exact render={props => <LoginPage {...props} login={this.login} />} />
					<Route path="/user-page" exact component={UserPage} />
					<Route path="/user-page/view-experiments/:id" exact component={ExperimentStat} />
					<Route path="/user-page/create-experiment" exact component={CreateExperimentForm} />
					<Route path="/user-page/create-experiment/:id(\d+)" exact component={CreateExperiment} />
					<Route path="/user-page/create-experiment/completed" exact component={CreateCompleted} />
					<Route path="/user-page/create-experiment/submit" exact component={CreateSubmission} />
					<Route path="/sign-up-page" exact component={SignUpPage} />
					<Route path="/forget-password" exact component={ForgetPassPage} />
					<Route path="/doctor-page" exact component={DoctorPage} />
					<Route path="/doctor-page/my-patients" exact component={MyPatients} />
					<Route path="/user-page/create-template" exact component={CreateTemplate} />
					<Route path="/blog" exact component={BlogPage} />
					<Route render={(props) => <HomePage {...props} aboutRef={this.state.aboutRef} />} />
				</Switch>
			</div>
		);
	}
}

export default withRouter(App);
