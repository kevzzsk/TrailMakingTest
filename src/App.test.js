import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import App from './App';
import HomePage from './components/HomePage'
import BlogPage from './components/BlogPage'
import PersonalParticulars from './components/PersonalParticulars'
import ExperimentTest from './components/ExperimentTest';
import BlogData from './template/BlogData'
import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Card} from '@material-ui/core/Card';
import { FormControl, TextField, CircularProgress } from "@material-ui/core"
import { MemoryRouter, Router } from "react-router-dom";
import {createMemoryHistory} from "history";

configure({adapter : new Adapter()});
function tick() {
  return new Promise(resolve => {
    setTimeout(resolve, 8000);
  });
}


describe("Routing", () => {
  it("404 error", ()=>{
    const wrapper = mount(
      <MemoryRouter initialEntries = {['/anyrandomnonexistantpage']}>
        <App/>
      </MemoryRouter>
    );
    expect(wrapper.find("h3").text()).toMatch("404");
  });

  it("navigates to blog page", () =>{
    const wrapper = mount(
      <MemoryRouter initialEntries = {['/blog']}>
        <App/>
      </MemoryRouter>
    );
    expect.assertions(2);
    expect(wrapper.find(BlogPage)).toHaveLength(1);
    expect(wrapper.find(HomePage)).toHaveLength(0);
  });

  it("navigates to blog page with simulated button click", async () =>{
    const wrapper = mount(
      <MemoryRouter initialEntries = {['/']}>
        <App/>
      </MemoryRouter>
    );
    const blogLink = wrapper.find('a#blog');
    blogLink.simulate('click', {button:0});  
    jest.setTimeout(30000);    
    await tick();
    expect.assertions(2);
    expect(wrapper.find(BlogPage)).toHaveLength(1);
    expect(wrapper.find(HomePage)).toHaveLength(0);
  });

});

describe("Home Page", () => {
  it("Home Page Snapshot", () =>{
    const tree = renderer.create(<HomePage/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Home Page loading state", async () => {
    const wrapper = mount(<HomePage/>);
    const submitBtn = wrapper.find('button#submitButton');
    expect(submitBtn).toHaveLength(1);
    submitBtn.simulate("submit");
    expect(wrapper.state("loading")).toEqual(true);
  });
  it("Home Page submission test error", async () => {
    const wrapper = mount(<HomePage/>);
    wrapper.setState({ id: 'test_id' });
    const fakeEvent = { preventDefault: () => {} };
    const experiment = wrapper.find("form#experiment_form");
    experiment.simulate("submit", fakeEvent);
    jest.setTimeout(30000);
    await tick(); //Delay for ten seconds to wait for response.
    wrapper.update();
    expect(wrapper.find(HomePage)).toHaveLength(1);
    expect(wrapper.state("error")).toMatch("valid ID");
   });

   it("Home Page submission success", async () => {
    const wrapper = mount(<HomePage/>);
    wrapper.setProps({history: []})
    wrapper.setState({ id: 'U0001' });
    const fakeEvent = { preventDefault: () => {} };
    const experiment = wrapper.find("form#experiment_form");
    experiment.simulate("submit", fakeEvent);
    jest.setTimeout(30000);
    await tick(); //Delay for ten seconds to wait for response.
    wrapper.update();
    expect(wrapper.prop("history")).toHaveLength(1);
   }); 
});

describe("blog page", () => {
  it("Rendering static blog page", ()=>{
    const tree = renderer.create(<BlogPage/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Number of blog pages in blog data", ()=>{
    expect(BlogData).toHaveLength(7);
  });

  it("Loading blog data", async ()=>{
    const wrapper = mount(<BlogPage/>);
    jest.setTimeout(30000);
    await tick();
    wrapper.update();
    let cardArr = wrapper.find("div.blogpost");
    console.log(cardArr);
    expect(cardArr).toHaveLength(BlogData.length);
    });
  }
);