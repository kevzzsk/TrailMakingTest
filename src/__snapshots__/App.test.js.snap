// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Home Page Snapshot 1`] = `
<div>
  <header
    className="header-bg"
  >
    <div
      className="head"
    >
      <h1>
        "Every 3 seconds someone in the world develops dementia"
      </h1>
      <div>
        <p>
          There were an estimated 50 million people worldwide living with dementia in 2017. This number will almost double every 20 years, reaching 75 million in 2030 and 131.5 million in 2050. Much of the increase will be in low and middle income countries. Already 58% of people with dementia live in low and middle income countries, but by 2050 this will rise to 68%. 
        </p>
      </div>
    </div>
  </header>
  <form
    id="experiment_form"
    onSubmit={[Function]}
  >
    <div
      className="form-group experiment"
    >
      <br />
      <div
        className="MuiFormControl-root w-100"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
            htmlFor="outlined-id"
            id="outlined-id-label"
          >
            Experiment ID
            <span
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
              â€‰
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="outlined-id"
              name="id"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="U001"
              required={true}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-338 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-339"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <small
        className="form-text text-muted"
      >
        By continuing you comply with our terms and conditions.
      </small>
      <div
        className="position-relative"
      >
        <button
          className="btn experiment-btn"
          disabled={false}
          id="submitButton"
          type="submit"
        >
          Try Experiment
        </button>
      </div>
    </div>
  </form>
  <section
    className="intro"
  >
    <h2>
      About Us
    </h2>
    <div>
      <p>
        We are a group of students from NTU who have developed an application based on Trail Making Tests for researchers and doctors alike to diagnose and treat patients with dementia. Researchers are able to use our application for research purposes by getting participants to use our application.
      </p>
    </div>
  </section>
  <div
    className="top-services"
  >
    <h2>
      We Are Looking For
    </h2>
    <div
      className="services"
    >
      <div
        className="service-one"
      >
        <p
          className="service-icon"
        >
          <i
            className="fas fa-users"
          />
        </p>
        <p
          className="service-title"
        >
          Participants
        </p>
        <p>
          Participants of research groups to aid in the research of dementia.
        </p>
      </div>
      <div
        className="service-two"
      >
        <p
          className="service-icon"
        >
          <i
            className="fas fa-user-graduate"
          />
        </p>
        <p
          className="service-title"
        >
          Researchers
        </p>
        <p>
          Scientists and researchers currently making great strides in identifying potential new ways to help diagnose, treat, and even prevent Alzheimer's and related dementias.
        </p>
      </div>
      <div
        className="service-three"
      >
        <p
          className="service-icon"
        >
          <i
            className="fas fa-user-md"
          />
        </p>
        <p
          className="service-title"
        >
          Doctors
        </p>
        <p>
          Doctors trained in brain conditions (neurologist) or doctors trained to treat older adults (geriatrician).
        </p>
      </div>
    </div>
  </div>
  <footer>
    <h3>
      TrailMakingTest by EasyA
    </h3>
    <p>
      Support 
      <br />
       TMT@easyA.com
    </p>
    <ul>
      <li>
        <a
          href="#"
        >
          <i
            className="fab fa-facebook-f"
          />
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          <i
            className="fab fa-twitter"
          />
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          <i
            className="fab fa-instagram"
          />
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          <i
            className="fab fa-github"
          />
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          <i
            className="fab fa-linkedin"
          />
        </a>
      </li>
    </ul>
  </footer>
</div>
`;

exports[`blog page Rendering static blog page 1`] = `
<div
  className="blog-bg"
>
  <div>
    <h3
      className="MuiTypography-root m-3 MuiTypography-h3"
    >
      Blog
    </h3>
    <div
      className="MuiSkeleton-root MuiSkeleton-text m-3 MuiSkeleton-animate"
      style={
        Object {
          "height": 460,
          "width": "97%",
        }
      }
    />
    <div
      className="MuiSkeleton-root MuiSkeleton-text m-3 MuiSkeleton-animate"
      style={
        Object {
          "height": 460,
          "width": "97%",
        }
      }
    />
    <div
      className="MuiSkeleton-root MuiSkeleton-text m-3 MuiSkeleton-animate"
      style={
        Object {
          "height": 460,
          "width": "97%",
        }
      }
    />
    )
  </div>
  <div
    className="m-2"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation1 p-4 MuiPaper-rounded"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4"
      >
        What can i do if someone i know is showing signs of dementia?
      </h4>
      <div
        className="MuiPaper-root MuiPaper-elevation0 MuiStepper-root MuiStepper-vertical"
      >
        <div
          className="MuiStep-root MuiStep-vertical MuiStep-completed"
        >
          <span
            className="MuiStepLabel-root MuiStepLabel-vertical"
          >
            <span
              className="MuiStepLabel-iconContainer"
            >
              <div>
                <i
                  className="far fa-edit"
                />
              </div>
            </span>
            <span
              className="MuiStepLabel-labelContainer"
            >
              <span
                className="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
              >
                He/she should see a doctor for an assessment and diagnosis.
              </span>
            </span>
          </span>
        </div>
        <div
          className="MuiStepConnector-root MuiStepConnector-vertical MuiStepConnector-completed"
        >
          <span
            className="MuiStepConnector-line MuiStepConnector-lineVertical"
          />
        </div>
        <div
          className="MuiStep-root MuiStep-vertical MuiStep-completed"
        >
          <span
            className="MuiStepLabel-root MuiStepLabel-vertical"
          >
            <span
              className="MuiStepLabel-iconContainer"
            >
              <div>
                <i
                  className="fas fa-info-circle"
                />
              </div>
            </span>
            <span
              className="MuiStepLabel-labelContainer"
            >
              <span
                className="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
              >
                Get some understanding about dementia. Your doctor can put you in touch with people who can help.
              </span>
            </span>
          </span>
        </div>
        <div
          className="MuiStepConnector-root MuiStepConnector-vertical MuiStepConnector-completed"
        >
          <span
            className="MuiStepConnector-line MuiStepConnector-lineVertical"
          />
        </div>
        <div
          className="MuiStep-root MuiStep-vertical MuiStep-completed"
        >
          <span
            className="MuiStepLabel-root MuiStepLabel-vertical"
          >
            <span
              className="MuiStepLabel-iconContainer"
            >
              <div>
                <i
                  className="fas fa-life-ring"
                />
              </div>
            </span>
            <span
              className="MuiStepLabel-labelContainer"
            >
              <span
                className="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
              >
                Rally for support. Caring for people with dementia is challenging and can be exhausting both physically and emotionally. Learn to talk about your problems. Make sure that you get enough rest. Take care of your own physical and mental well-being.
              </span>
            </span>
          </span>
        </div>
        <div
          className="MuiStepConnector-root MuiStepConnector-vertical MuiStepConnector-completed"
        >
          <span
            className="MuiStepConnector-line MuiStepConnector-lineVertical"
          />
        </div>
        <div
          className="MuiStep-root MuiStep-vertical MuiStep-completed"
        >
          <span
            className="MuiStepLabel-root MuiStepLabel-vertical"
          >
            <span
              className="MuiStepLabel-iconContainer"
            >
              <div>
                <i
                  className="fas fa-users"
                />
              </div>
            </span>
            <span
              className="MuiStepLabel-labelContainer"
            >
              <span
                className="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
              >
                Group support is important. Meet with others who are also caring for people with dementia. Sharing provides mutual support for caregivers.
              </span>
            </span>
          </span>
        </div>
        <div
          className="MuiStepConnector-root MuiStepConnector-vertical MuiStepConnector-completed"
        >
          <span
            className="MuiStepConnector-line MuiStepConnector-lineVertical"
          />
        </div>
        <div
          className="MuiStep-root MuiStep-vertical MuiStep-completed"
        >
          <span
            className="MuiStepLabel-root MuiStepLabel-vertical"
          >
            <span
              className="MuiStepLabel-iconContainer"
            >
              <div>
                <i
                  className="fas fa-user-md"
                />
              </div>
            </span>
            <span
              className="MuiStepLabel-labelContainer"
            >
              <span
                className="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
              >
                Get expert advice especially with coping with challenging behaviour.
              </span>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiPaper-elevation1 p-4 mt-3 card-left MuiPaper-rounded"
    >
      <i
        className="fas fa-phone-alt align-self-center"
        style={
          Object {
            "color": "rgb(248,107,70)",
            "fontSize": "40px",
          }
        }
      />
      <div
        className="ml-2"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5"
        >
          HPB Dementia Infoline
        </h5>
        <h3
          className="MuiTypography-root MuiTypography-h3"
          style={
            Object {
              "color": "rgb(248,107,70)",
              "fontWeight": "600",
            }
          }
        >
          1800 223 1123
        </h3>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiPaper-elevation1 p-4 mt-3 card-left MuiPaper-rounded"
    >
      <i
        className="fas fa-phone-alt align-self-center"
        style={
          Object {
            "color": "rgb(248,107,70)",
            "fontSize": "40px",
          }
        }
      />
      <div
        className="ml-2"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5"
        >
          Alzheimer's Disease Association Dementia Helpline
        </h5>
        <h3
          className="MuiTypography-root MuiTypography-h3"
          style={
            Object {
              "color": "rgb(248,107,70)",
              "fontWeight": "600",
            }
          }
        >
          6377 0700
        </h3>
      </div>
    </div>
  </div>
</div>
`;
